src/Microsoft.Dnx.Runtime/Loader/PackageAssemblyLoader.cs

```diff
@@ -28,7 +28,19 @@ public Assembly Load(AssemblyName assemblyName, IAssemblyLoadContext loadContext
         {
             // TODO: preserve name and culture info (we don't need to look at any other information)
             string path;
-            if (_assemblies.TryGetValue(new AssemblyName(assemblyName.Name), out path))
+            var newAssemblyName = new AssemblyName(assemblyName.Name);
```

Comment:
Even if i pass the assemblyName as is and fix the comparer, the entries are not being matched. I had to create an assemblyName without neutral culture for comparer to compare.